homeassistant/components/binary_sensor/bayesian.py

```diff
@@ -120,40 +126,41 @@ def async_threshold_sensor_state_listener(entity, old_state,
             if new_state.state == STATE_UNKNOWN:
                 return
 
-            entity_obs = self.entity_obs[entity]
-            platform = entity_obs['platform']
+            entity_obs_list = self.entity_obs[entity]
 
-            self.watchers[platform](entity_obs)
+            for entity_obs in entity_obs_list:
+                platform = entity_obs['platform']
 
-            prior = self.prior
-            for obs in self.current_obs.values():
-                prior = update_probability(prior, obs['prob_true'],
-                                           obs['prob_false'])
+                self.watchers[platform](entity_obs)
 
-            self.probability = prior
+                prior = self.prior
+                for obs in self.current_obs.values():
+                    prior = update_probability(prior, obs['prob_true'],
+                                               obs['prob_false'])
+                self.probability = prior
 
-            self.hass.async_add_job(self.async_update_ha_state, True)
+                self.hass.async_add_job(self.async_update_ha_state, True)
```

Comment:
This entity only needs to be updated after the new probability is calculated, ie outside the loop.

Reply:
Thanks for catching this, got carried away with the indenting, and all the tests passed so I guess I didn't notice 😱